data/logs_model/logs_producer/splunk_logs_producer.py

```diff
@@ -22,19 +23,50 @@ def __init__(
         url_scheme="https",
         verify_ssl=True,
         index_prefix=None,
+        ssl_ca_path=None,
     ):
+        connect_args = {
+            "host": host,
+            "port": port,
+            "token": bearer_token,
+            "scheme": url_scheme,
+            "verify": verify_ssl,
+        }
+        if ssl_ca_path:
+            # Create an SSLContext object for cert validation
+            context = ssl.create_default_context()
+            context.check_hostname = True
+            context.verify_mode = ssl.CERT_REQUIRED
+            if not verify_ssl:
+                context.check_hostname = False
+                context.verify_mode = ssl.CERT_NONE
+            try:
+                context.load_cert_chain(certfile=ssl_ca_path)
+            except ssl.SSLError as e:
+                logger.exception("SSL certificate error %s", e)
+                raise e
+            except FileNotFoundError as e:
+                logger.exception("Path to cert file is not valid %s", e)
+                raise e
```

Comment:
Usually it's a bad idea to do both: log an error and propagate it to the caller. How does the caller handle these errors?